eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -4,44 +4,72 @@ parameters:
+          $results = @()
+          echo "##vso[task.setvariable variable=HasChanges]$false"
+          $shaBeforePatches = git rev-parse --short HEAD
+          foreach ($file in (Get-ChildItem $(PatchFilesLocation)))
+          {
+             Write-Host $file.FullName
+             git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" am -3 $file.FullName
+             if ($lastExitCode -ne 0) { 
+               git am --show-current-patch=diff
+               Write-Error "##vso[task.LogIssue type=warning;]Failed to properly apply patch files to [https://github.com/azure/${{ repo }}]"
+               exit 1
+             }
+          }
+          $shaAfterPatches = git rev-parse --short HEAD
+          $hasChanges = $shaBeforePatches -ne $shaAfterPatches
+          echo "##vso[task.setvariable variable=HasChanges]$hasChanges"
+        }
+        else
+        {
+          if (!(Test-Path $repoPath)) { mkdir $repoPath }
+          Remove-Item -v -r $repoPath
+          Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
+          Get-ChildItem -r $repoPath
+        }
+      displayName: Copy or Apply Patch for ${{ parameters.DirectoryToSync }} from azure-sdk-tools to ${{ repo }}
       workingDirectory: $(System.DefaultWorkingDirectory)
 
-    - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
-      parameters:
-        RepoName: ${{ repo }}
-        RepoOwner: Azure
-        PRBranchName: ${{ parameters.PRBranchName }}
-        BaseBranchName: ${{ parameters.BaseBranchName }}
-        CommitMsg: ${{ parameters.CommitMessage }}
-        PRTitle: ${{ parameters.PRTitle }}
-        PRBody: ${{ parameters.PRBody }}
-        PRLabels: "Central-EngSys, EngSys"
-        PushArgs: -f
-        WorkingDirectory: $(System.DefaultWorkingDirectory)/${{ repo }}
-        ScriptDirectory: $(System.DefaultWorkingDirectory)/eng/common/scripts
-
-    - pwsh: |
-        $PRData = "Azure;${{ repo }};$(Submitted.PullRequest.Number)"
-        Add-Content -Path ${{ parameters.PRDataArtifactPath }} -Value $PRData
-      displayName: Write Sync PR Data to Artifact File
+    - task: PowerShell@2
+      displayName: Push changes
       condition: and(succeeded(), eq(variables['HasChanges'], 'true'))
+      inputs:
+        pwsh: true
+        workingDirectory: $(System.DefaultWorkingDirectory)/${{ repo }}
+        filePath: ${{ parameters.ScriptDirectory }}/git-branch-push.ps1
+        arguments: >
+          -PRBranchName "${{ parameters.PRBranchName }}"
+          -CommitMsg "${{ parameters.CommitMessage }}"
+          -GitUrl "https://$(azuresdk-github-pat)@github.com/${{ parameters.PROwner }}/${{ repo }}.git"
+          -PushArgs "${{ parameters.PushArgs }}"
+          -SkipCommit "${{parameters.SkipCheckingForChanges}}"
+
+    - task: PowerShell@2
+      displayName: Queue test pipeline
+      condition: and(succeeded(), ne(variables['${{repo}}-template-definition-id'], ''))
```

Comment:
Where do these definition id's get setup? Is it only in the pipeline itself? 

Should this be part of the sync-directory.yml file or the eng-common-sync.yml? It feels like it should be part of the eng-common-sync.yml file but I think we are starting to make this sync-directory.yml file be less useful for anything else so maybe it doesn't matter.

Reply:
I did not want to dump the pipeline definition IDs in the source. I know you weren't ok with that in some of my previous work, so I put it in the pipeline via the UI.

Reply:
I don't particularly like them in either location :) However I was mostly curious about where this lives. Is there a way to look up the definition id by name? So we could build a name and then try to find the associated definition.

Reply:
It looks like one still requires the Definition ID to request information about the pipeline. https://docs.microsoft.com/en-us/rest/api/azure/devops/pipelines/pipelines/get?view=azure-devops-rest-6.1

Reply:
Sadly the API you need is split across multiple groups of REST API, you can use this one to look up a pipeline definition by name:

https://docs.microsoft.com/en-us/rest/api/azure/devops/build/definitions/list?view=azure-devops-rest-6.1

Reply:
@chidozieononiwu can you please file a follow-up item to figure out a better way to configure the list of test pipelines we want to trigger? I think I'd prefer a list of pipelines in the yml file over having them be hidden away in the variables tab in the web UI.